0xPolygonMiden/miden-client
# feat: Update load genesis command for newer genesis and account files

Addresses #86 

On [#138](https://github.com/0xPolygonMiden/miden-node/pull/138) from miden node we're adding a config file for generating the genesis block. Now it's possible to create many accounts from the get go. The client thus needs to be able to read the newer files. This is:

- the `genesis.dat` file which contains a serialized GenesisState (this is the same as before)
- the `account_{i}.mac` files which contain a serialized AccountData each. 

Here we assume the path provided to the load-genesis command points to a directory that contains both the `genesis.dat` file and a subdirectory `accounts` containing all of the `account_{i}.mac` files within it.

### Summary of changes

- new parsing of genesis data
- refactor: moved load genesis logic into the client. I think this will be useful for testing purposes by users of the client that don't use the CLI. Also made testing the load of the genesis a bit easier
- added test for loading genesis data

### Testing changes

To test it you can:

- clone the [miden node repo](https://github.com/0xPolygonMiden/miden-node), `cd` into it
- if wanted, you can add more accounts to `node/genesis.toml`
- generate the genesis data with `cargo run --bin miden-node --features testing -- make-genesis --config node/genesis.toml`
- start the node with `cargo run --bin miden-node --features testing -- start --config node/miden.toml` (this step shouldn't be necessary, the issue is being fixed in #85)
- now `cd` to the client's directory
- load the genesis data with `cargo run --features testing,concurrent load-genesis --genesis-path <path_to_the_miden_node_repo>`
- you can also check that the accounts were loaded with `cargo run --features testing,concurrent account list`
- you can also do a mint transaction with `cargo run --release --features testing,concurrent -- transaction new mint <dst_account_id> <faucet_account_id> 10`. Use the accounts you saw from the previous step.
